pandas/core/sparse/frame.py

```diff
@@ -540,9 +540,6 @@ def _combine_frame(self, other, func, fill_value=None, level=None):
         this, other = self.align(other, join="outer", level=level, copy=False)
         new_index, new_columns = this.index, this.columns
 
-        if self.empty and other.empty:
-            return self._constructor(index=new_index).__finalize__(self)
```

Comment:
this is covered by a test?

(in general, I would rather leave sparse alone, this is deprecated anyway)

Reply:
Yes it is.

I understand the impulse to leave sparse alone, but ATM some of it is a barrier to simplifying ops code

Reply:
As I was just saying on the other PR as well, there is also the option to just leave sparse as is, but still go forward with simplifying the ops code for non-sparse

Reply:
This can also make it easier for you to focus only on the ops code for normal frame (without additional complexity of sharing with sparse)

Reply:
Not sure I understand.  core.ops functions call the DataFrame/SparseDataFrame methods, so anything I want to do to change the behavior of the non-sparse methods needs to be reflected in the sparse methods